lib/puppet/transaction/report.rb

```diff
@@ -176,7 +178,7 @@ def add_resource_status(status)
 
   # @api private
   def compute_status(resource_metrics, change_metric)
-    if resources_failed_to_generate || (resource_metrics["failed"] || 0) > 0
+    if resources_failed_to_generate || (resource_metrics["failed"] || 0) > 0 || @logs.any? { |lg| lg.level == :err }
```

Comment:
yes, very much needed - see the commit comment. There are additional things that happen after the report is finalized.